File: llm_generated_code/ChatGPT/Python/Level2/Q6.py
def find_liars(claims):
    n = len(claims)
    
    # Sort the claims
    claims.sort()
    
    for liars in range(n + 1):
        count = sum(1 for claim in claims if claim >= liars)
        if count == liars:
            return liars
    
    return -1  # Contradiction

def main():
    t = int(input())
    results = []
    
    for _ in range(t):
        n = int(input())
        claims = list(map(int, input().split()))
        result = find_liars(claims)
        results.append(result)
    
    for result in results:
        if result == -1:
            print("Contradiction")
        else:
            print(result)

if __name__ == "__main__":
    main()

File: llm_generated_code/ChatGPT/CPP/Level2/Q6.cpp
#include <iostream>
#include <vector>
#include <algorithm>
using namespace std;

int find_liars(vector<int>& claims) {
    int n = claims.size();
    
    // Sort the claims
    sort(claims.begin(), claims.end());

    for (int liars = 0; liars <= n; ++liars) {
        int count = 0;
        for (int claim : claims) {
            if (claim >= liars) {
                count++;
            }
        }
        if (count == liars) {
            return liars;
        }
    }
    
    return -1;  // Contradiction
}

int main() {
    int t;
    cin >> t;
    vector<int> results;
    
    while (t--) {
        int n;
        cin >> n;
        vector<int> claims(n);
        for (int i = 0; i < n; ++i) {
            cin >> claims[i];
        }
        int result = find_liars(claims);
        if (result != -1) {
            results.push_back(result);
        } else {
            results.push_back(-1); // Representing "Contradiction"
        }
    }
    
    for (int result : results) {
        if (result == -1) {
            cout << "Contradiction" << endl;
        } else {
            cout << result << endl;
        }
    }
    
    return 0;
}
